### DISCUSSION TITLE
Keine Verbindung zur Pylontech Batterie

### DISCUSSION THREAD

**Discussion by gruenteefan** [hoylabs/OpenDTU-OnBattery#1416] [Q&A] [answered]
Hallo zusammen, 

ich bekomme über die Netzwerkschnittstelle meiner OpenDTU on Battery keine Verbindung zur Pylontech US3000C. 
Ich nutze dabei das dem Pylontech beiliegende Kabel Battery an Inverter und stecke es am Pylontech am ACAN Anschluss an.

Woran könnte das liegen?

**Comment by Solarteur** [comment]
Ohne Angabe, welche OpenDTU-OnBattery-Hardware du verwendest,  ist aus meiner Sicht hier keine Hilfe möglich.

**Comment by Manos1966** [comment]
> Ohne Angabe, welche OpenDTU-OnBattery-Hardware du verwendest, ist aus meiner Sicht hier keine Hilfe möglich.

Ist richtig, poste bitte aus welcher Hardware deine OpenDTU besteht.

Probiere ein normales LAN RJ-45 Netzwerk-Kabel (das dem Pylontech beiliegende Kabel Battery an Inverter funktioniert nicht)

**Reply by gruenteefan** [reply]
Herzlichen Dank! Das probiere ich aus. Ist die korrekte Baud-Rate 9600 oder 115000?

**Comment by gruenteefan** [comment]
Hallo, danke für eure Antworten und entschuldigt bitte meine fehlenden Angaben!

Ich habe diese DTU hier gekauft:
https://www.kleinanzeigen.de/s-anzeige/opendtu-onbattery-pcb-platine-esp32-s3-plug-play/2802534023-168-2562


Ich bin für jegliche Hilfe sehr sehr dankbar!

**Reply by Manos1966** [reply]
schoene Platine 👍

**Reply by schlimmchen** [reply]
SIEBENUNDVIERZIG EURO **ohne** ESP32, geschweige denn ESP32-S3, geschweige denn NRF24, und schon gar kein Platz für ein CMT2300A? Ich bin sprachlos. Und ihr kauft das? Obwohl wir überall und ständig sagen: "Kauf ein OpenDTU Fusion Board"? Check ich nicht. Hab ich irgendetwas missverstanden, warum das das Killer-Board ist?

**Reply by Manos1966** [reply]
Es ist nicht unbedingt notwendig, Menschen, die bereits etwas gekauft haben, den Eindruck zu vermitteln, dass sie den größten Fehler ihres Lebens begangen haben...

**Reply by Sorbat** [reply]
Auf der PCB ist doch nichts drauf... Ich lerne gerade ein wenig kicad und bin erschrocken darüber, wie günstig man in China pcbs bestellen kann. Meine letzten 5 Stück haben 4,80€ gekostet, inkl. Versand. Das waren ca 95cent pro Platine - unbestückt natürlich. 
Aber was hat er denn drauf? Mit das  teuerste werden wohl die adums sein, lcsc 97cent. 
Ohne jetzt genau zu wissen welche Buchsen das sind, die dort verbaut sind (vielleicht kosten die ja auch 30€, ich denke aber eher nicht), möchte ich mal behaupten, dass man die pcb in kleinen Stückzahlen, also 5 Stück, vermutlich für unter 10€/stück selber bauen könnte. Und Lukas baut ja offensichtlich keine kleinen Stückzahlen davon, da er die Dinger am laufenden Band bei Kleinanzeigen anbietet.

Also, war es der größte Fehler des Lebens? Nein.
War es eine gute Entscheidung? Nein. 


... man hätte doch zumindest den s3 als Smd mit drauf packen können, USB-c Buchse dazu, die buchsenleisten gespart und schon hätte das ganze nen anderen Charakter gehabt. Natürlich auch dann nicht für 47€.

**Reply by schlimmchen** [reply]
> Es ist nicht unbedingt notwendig, Menschen, die bereits etwas gekauft haben, den Eindruck zu vermitteln, dass sie den größten Fehler ihres Lebens begangen haben...

Das ist richtig. Es war auch nicht meine Absicht jemanden dumm aussehen zu lassen. Man macht halt Fehler, das tun wir alle. Aber mir ist es schon wichtig meine Meinung dazu zu äußern. Es ist natürlich auch so, dass meine Warnung vermutlich hier nicht gelesen wird, insofern hätte ich es mir vielleicht wirklich sparen können... Ich frag mich, wie ich Leute davon überzeugen kann, eine bessere Entscheidung zu treffen. Die Doku verträgt kaum noch mehr Hinweise, ohne dass es aufdringliche Werbung würde...

**Reply by Manos1966** [reply]
Die Menschen die hier waren (github.com/hoylabs/OpenDTU-OnBattery) haben keine Schwierigkeiten eine Platine, oder selber etwas zusammenzubasteln.

Die Menschen die Fertige "Loesungen" kaufen, haben nicht vorher gewusst, dass solche Loesungen existieren.
Wenn man die Anzahl verkauften Hoymiles Wechselrichter gegen der Zahl installierten AHOY, OpenDTU, OpenDTU-OnBattery gegenueber stellt, bekommt man eine Idee, wieviel Leute nicht davon wissen.
Jeder Mensch, der eine solche "Loesung" kauft, wird automatisch Mitglied der Hoylabs Community = Unbezahlbar!

Ich muss immer an den Spruch denken: "learn to see the value of things, not the price"

Insofern bleibe ich bei meiner Aussage: Unbezahlbar!

PS. Ich habe auch selber vor drei Jahren 45EUR fuer meine erste ESP8286 AHOY DTU  bezahlt. Verglichen  zu den 30.000 EUR die ich damals bezahlen haette koennen (bei einem namhaften Solar Anbieter), es waren die besten 45EUR meines Lebens!

**Reply by Sorbat** [reply]
Vielleicht ist die Doku auch einfach auch nicht der optimale Ort um das Fusion zu empfehlen. Sie ist sehr umfangreich und erschlägt einen gerne mal. Ich kann mir nicht vorstellen, dass sich jeder durch die Doku kämpft bevor er irgendwas kauft, häufig wird es wohl anders herum sein.

Den weiteren Verkauf dieses bestimmten Board sollte man aber relativ leicht unterbinden können. Der Anbieter erfüllt keinerlei gesetzliche Auflagen, er bietet hier die von Ihm produzierten Pcbs als Privatverkäufer an. Keine Registrierung nach dem ElektroG, keine Angaben entsprechend des ProdSG, Entzug der gesetzlichen Pflichten gegenüber dem Verbraucher und derer Rechte und und und 

@Manos1966 Es wollte dir jemand einen ESP für 30k anbieten? ein glück, dass du den nicht genommen hast

**Reply by Manos1966** [reply]
> Es wollte dir jemand einen ESP für 30k anbieten? ein glück, dass du den nicht genommen hast

Oh Mann @Sorbat 🙄 
Es war ein "Standard Solar Anlage" Angebot, damals haben alle Angebote 28.000EUR und mehr gekostet.

**Comment by Solarteur** [comment]
Der Vorschlag von @Manos1966, ein normales RJ-45_Netzwerk-Kabel zu verwenden, ist richtig und wichtig.
Laut https://github.com/hoylabs/OpenDTU-OnBattery/wiki/Pylontech-battery und den Angaben von Lukas bei Kleinanzeigen sollten CAN-H und CAN-L damit richtig angeschlossen sein.
Zur Baud-Rate: Ich habe bei mir alle DIP-Schalter des Pylontech-Akkus auf "Null" (nach oben) belassen.
Wenn das so nicht funktioniert, poste bitte einen Screenshot deiner "Batterie-Einstellungen".

**Comment by gruenteefan** [comment]
Leider habe ich noch immer keine Lösung gefunden. Die Werte des Pylontech-Akkus erscheinen nicht in der DTU. 

Ich habe einen Screenshot meiner Batterie-Einstellungen und der Hardware-Einstellungen gemacht. 

Woran könnte das liegen? Die PIN-Belegung des Netzwerkkabels ist mir soweit klar (Handbuch des Pylontech-Akkus). Wie müsste die PIN-Belegung des RJ45-Steckers an der DTU sein?


![Screenshot 2025-01-28 074924](https://github.com/user-attachments/assets/6663f79c-c5f2-4abf-92f9-b87b85daa98a)
![Screenshot 2025-01-28 074851](https://github.com/user-attachments/assets/8576d9da-56bc-4828-9f32-e0bef9b80d14)

**Reply by Manos1966** [reply, marked as answer]
Also....
du hast (Got sei Dank) uns den Link gegeben, woher du deine OpenDTU gekauft hast 👍 
Hast du da auch das zweite Bild gesehen? (siehe unten)
![image](https://github.com/user-attachments/assets/1513ff57-9284-478e-b6df-75b05e999606)

Da siehst du
![image](https://github.com/user-attachments/assets/50ace5f5-f219-4aab-bab4-48d42414e2bb)

die auf die GPIO eines ESP32-S3 gehen:
![image](https://github.com/user-attachments/assets/436a610e-2b7e-43a2-8245-a5b0fbbf1084)

... die "Meilen entfernt" sind von den 12 u. 13 die du auf deine PIN-Belegung definiert hast 😜 

Ich kann nur **vermuten**:
Wenn du dein Profil auf die Version "ESP32 mit 2 Victron (MPPT) und cmt2300a (blinkyparts)" umstellst (ohne SHUNT) es wird funktionieren. 🤞

**Comment by gruenteefan** [comment]
Hallo Manos, 

Du hast mit allem Recht! Es hat tatsächlich am falschen Profil gelegen. 

Ich danke Dir sehr für Deine Mühe und Deine Hilfsbereitschaft!

Viele Grüße
Stefan

**Comment by gruenteefan** [comment]
Ich hab das Gefühl, dass ich jetzt auf der Zielgeraden bin; leider funktioniert die Ansteuerung des MPPTs noch nicht (Victron Smartsolar 150/35).

Sobald ich das VE.direct-Kabel (angeschlossen am MPPT) in die DTU einstecke, friert die DTU ein und der Wechselrichter geht aus. Entferne ich das Kabel wieder, läuft wieder alles korrekt. 

Woran könnte denn das jetzt noch liegen?

![Screenshot 2025-01-28 121614](https://github.com/user-attachments/assets/81c3c099-0ebf-4672-8312-0d70d47c50d9)

**Comment by Solarteur** [comment]
Welches Modul zur Verbindung zwischen Wechselrichter und openDTU verwendest du? Das Pin-Mapping ist für ein CMT2300a-Modul (HM**S**-Wechselrichter) eingerichtet, die Anschlüsse auf der Platine sprechen eher für ein nRF24L01+-Modul (HM-Wechselrichter).

Edit: Ich habe mit Victron keinerlei Erfahrung. Aus meiner Sicht passen die GPIO aus dem Schaltplan und die aus dem Pin-Mapping unter "Victron" nicht zusammen.

**Comment by gruenteefan** [comment]
Ich habe einen HMS-2000-4T Wechselrichter, in der DTU ist ein CMT2300A Modul verbaut. Alles läuft jetzt eigentlich, bis auf die Steuerung des Victron Ladereglers

**Reply by Solarteur** [reply]
Ja ok. Von Blinkyparts gibt es eine Adapterplatine für ein CMT2300A-Modul, das passt.

Die beiden Schaltkreise unter Victron im Schaltplan sind wohl ADuM1201 als Zweikanal-Digital-Isolatoren. Und das Pin-Mapping könnte stimmen - ich habe jetzt leider keine Idee mehr.

**Comment by gruenteefan** [comment]
Ich habe jetzt auf Anraten von Lukas (Verkäufer der DTU) die Pins getauscht (46 gegen 45, 13 gegen 12). Leider immer noch das gleiche Resultat (DTU hängt sich auf, ist per IP nicht mehr erreichbar und der Wechselrichter geht aus). Kann hier evtl. jemand helfen?

**Reply by Manos1966** [reply]
On mein Got @gruenteefan 

Stand der Dinge, wir sind vom "Keine Verbindung zur Pylontech Batterie" zu "Nichts funktioniert, ich habe keine Ahnung und da wo ich die Platine her habe genau so wenig"

Ich fange an die OpenDTUFusion Platine mehr und mehr zu schaetzen...

Sorry falls ich etwas "direkt" bin, ich habe Allergie gegen "Salami Taktik" 

ALSO, was FUNKTIONIERT heute bei deiner DTU? Mach bitte eine Liste was du hast und was nicht funktioniert.

**Comment by gruenteefan** [comment]
Oh, das ist jetzt aber eine harte Aussage, finde ich. Ich arbeite mich halt von einer Baustelle zur nächsten. 

Was ich habe:

Ich habe mich an diesem Aufbau orientiert:

![Aufbau](https://github.com/user-attachments/assets/a64c3286-28c7-4d66-98d9-25c5df30da58)

Ich habe keine Raspberry, sondern eben eine OpenDTU on Battery, die ich fertig von Lukas gekauft habe. 

6 Solarmodule, 3 x 2 in Reihe
Akku: Pylontech US3000C
Wechselrichter: HMS-2000-4T
Laderegler: Victron Smartsolar 150/35
Smartmeter: Shelly Pro 3EM
verbunden über zwei Busbars. Als Sicherungen habe ich Neozed Sicherungen verwendet. 
Ich habe auch noch einen Cerbo GX.

Was nicht funktioniert: 
Der Victron Smartsolar Laderegler muss jetzt noch an die OpenDTU on Battery angeschlossen werden. Das funktioniert jetzt eben nicht, beim Anschluss des Ladereglerst mittels ve-direct-Kabel hängt sich die DTU auf und sie ist per IP nicht mehr erreichbar und der Wechselrichter geht aus.

Vielen Dank schonmal für die Hilfe.

**Reply by Manos1966** [reply]
@gruenteefan Danke fuer die gute Beschreibung.

Ich verstehe es also wie folgt:
- Pylontech Verbindung: Funktioniert
- Stromzaehler Verbindung: funktioniert
- HMS-2000 Verbindung: funktioniert
- Victron Verbindung: Funktioniert NICHT



Du hast (nur) einen Victron.
Daraufhin stelle bitte grundsätzlich:
rx2 -1
rx3 -1
tx2 -1
tx3 -1
... damit dein System nur um die Pins tx & rx kümmern muss
 
laut Plan 
![image](https://github.com/user-attachments/assets/e197dc23-d968-4e77-88e1-7d258e58c614)

und die GPIOs in Frage 45 & 46 (wie du bereits gesagt hast)
![image](https://github.com/user-attachments/assets/3b781e14-0e51-49ce-8629-5b73ae135a29)

Ich gehe davon aus du benutzts die "ESP32 mit 2 Victron (MPPT) und cmt2300a (blinkyparts)"
Bitte stelle rx2 und tx2 auf -1:
        "victron": {
            "rx": 45,
            "tx": 46,
	    "rx2": -1,
            "tx2": -1

stelle auch 
        "display": {
            "type": -1,
            "data": -1,
            "clk": -1

(du hast kein Display gehe ich davon aus)


**Danach kommst es auf dein VE.Direkt Kabel an:
Was fuer ein Kabel zwischen Victron und OpenDTU benutzt du?
Welche Adern sind angeschlossen (falls du es selber gebaut hast)
Welcher Marke (falls du es nicht selber gebaut hast)**

Deine Platine hat "galvanische Trennung"  = eine ADUM Platine zwischen ESP32 und Victron.
![image](https://github.com/user-attachments/assets/e197dc23-d968-4e77-88e1-7d258e58c614)

Diese Platine braucht neben tx & rx (von Victron-Seite) auch GND und VCC

Zusammenfassung:
- Hast du ein **fertiges** Kabel gekauft, dann sind alle vier Adern angeschlossen und die rx/tx wird im Kabel selbst "gekreuzt", 
LOESUNG: du muss nur die tx/rx ueberpruefen ob sie korrect ankommt: "rx": 45, "tx": 46 oder "rx": 46, "tx": 45 in der Konfiguration ist die Loesung. Du muss beide Varianten testen.
- Hast du das Kabel **selber** gemacht, **pruefe ob du die Adern GND und VCC (5V Strom) angeschlossen hast.** Ich gehe stark davon aus, dass du die tx/rx Ader  nicht angefasst (="gekreuzt") hast , daher die Konfiguration "rx": 45, "tx": 46  die richtige ist.

Dieses Bild hilft. Pass gut auf wie der Stecker ist auf Basis der "offenen Markierung" (im Bild ist sie Oben). Bei manchen Victron MPPTs ist sie genau umgekehrt. Machst du einen Fehler, ist tx/rx umgekehrt, noch schlimmer du hast GND und VCC 5+ **umgekehrt**! und danach schreikt dein ESP32
![image](https://github.com/user-attachments/assets/5b2bd894-3390-4a51-a2be-a8abae369326)


Viel Glueck! 🤞

**Comment by gruenteefan** [comment]
@Manos1966 

Vielen herzlichen Dank für Deine Mühe!

Habe das alles so umgesetzt, wie von Dir empfohlen. Leider kein Erfolg :-(

Sobald ich das VE-Direct-Kabel anschließe, gehen die LEDs vom Victron aus. Und im Nachgang fällt dann der hoymiles WR aus. 

Das Kabel ist original von Victron, TX und RX sind gekreuzt.

**Comment by gruenteefan** [comment]
Habe testweise jetzt auch Power+ und GND getauscht; gleiches Resultat!

Wie müssen eigentlich die Einstellungen am Victron Laderegler sein? Hab ihn auf Werkseinstellungen zurückgesetzt und nur die Batterie auf 48V geändert. Die Batterie ist aktuell noch (per Schalter) von der Busbar weggeschaltet.

**Comment by Manos1966** [comment]
@gruenteefan Hmmm.....

Ist das deine Platine? 
- welcher ist der Victron Anschluss den du nutzts? (im Bild habe ich 1 oder 2 markiert) 
- weiss du welche Zuordnung sie haben? (welcher ist gesteuert vom VICTRON1 transistor und welcher vom VICTRON2, die Transtoren mitten im Bild)

wofuer sind diese Jumpers (mit Kreis markiert)

![image](https://github.com/user-attachments/assets/08c787d2-8943-47d0-b51f-867d5ec83219)

> Das Kabel ist original von Victron, TX und RX sind gekreuzt.

Sehr gut, also deine Einstellung muss "rx": 46, "tx": 45 sein.


> Habe testweise jetzt auch Power+ und GND getauscht; gleiches Resultat!

Jetzt muss ich fragen, wie hast du das gemacht? Ich dachte du hast ein Victron Kabel, es ist eingeschweißt. 

> Wie müssen eigentlich die Einstellungen am Victron Laderegler sein? Hab ihn auf Werkseinstellungen zurückgesetzt und nur die Batterie auf 48V geändert. 
Du hast einen Smartsolar 150/35, du stellst ihn ein ueber die App (Manual lesen). 
Hast du keine APP sollte der Switch auf Position 7 sein (LiFePo4 Batterien)

> Die Batterie ist aktuell noch (per Schalter) von der Busbar weggeschaltet.

Hat der Victron "Strom" (entweder von der Batterie, oder von den Solar Paneelen)? Eins von den zwei muss es sein (ansonsten wie soll er funktionieren?)

**Comment by gruenteefan** [comment]
Nein, meine Platine sieht etwas anders aus. Die Jumper sind nicht vorhanden. 

Hier ein Bild:
![PXL_20250130_163744789](https://github.com/user-attachments/assets/c841b70d-f283-49eb-8c02-2bc4c120aba9)

Der Victron ist über die DC Sammelschiene (jeweils + und - separat) sowohl an den Hoymiles Wechselrichter, als auch an die PV-Module angeschlossen:

![Aufbau](https://github.com/user-attachments/assets/4ea6ccdb-3845-4b66-8aed-47310939af73)

Zum Kabel: Ich habe den Mantel aufgetrennt, mit einer Nadel Belegung 1&4 herausgefunden, die Isolierung entfernt und den Kontakt mittels einer Wago Verbindungsklemme neu hergestellt.

Danke für Deine Hilfe!

**Reply by Manos1966** [reply]
Verdamt! 🤦‍♂️ 

welcher Eingang (Links) ist Victron1 ?
![image](https://github.com/user-attachments/assets/56f14d59-1405-4926-8b88-843b87cacab7)

**Reply by Manos1966** [reply]
Ich hoffe du hast ein Multimeter.

Wenn ich den Schaltplan sehe, aufgrund der C1 und C2 Widerstände kann ich die VOA (2), VIB (3), VIA (7) und VOB (6) identifizieren (auf dem Bild markiert)

Mit dem Multimeter (Ohm Messung) pruefe bitte ob 

- VE.Direkt Stecker PIN2 korrekt angeschlossen ist (sollte entweder mit VOA 7 oder VIA 2 verbunden sein)
- VE.Direkt Stecker PIN3 korrekt angeschlossen ist (sollte entweder mit VOB 7 oder VIB 2 verbunden sein)

Danach, noch wichtiger, pruefer bitte ob
- ESP32-S3 GPIO46 korrekt angeschlossen ist (sollte entweder mit VOB 6 oder VIB 3 verbunden sein)
- ESP32-S3 GPIO45 korrekt angeschlossen ist (sollte entweder mit VOA 7 oder VIA 2 verbunden sein)

![image](https://github.com/user-attachments/assets/3ca6a53c-fa34-4e12-a2c5-6011b9c294ac)

**Comment by gruenteefan** [comment]
@Manos1966 
Vielen Dank für Deine Hilfe. Ich bin noch am Durchmessen. Gar nicht so einfach - sobald der ESP draufsteckt, komme ich natürlich nicht mehr an die die Kontakte VOB/ VOA. Ich melde mich dazu in Kürze wieder!

Ich habe den Akku heute auch ohne Victron Schnittstelle erstmals laden können (98%). 

Aus irgendeinem Grund hat der Victron 150/35 aber eine Art "Begrenzung" (knapp unter 800 Watt) gehabt. Die 6 Panels hätten heute deutlich mehr als 680 Watt liefern müssen. 

Schon seltsam, oder?

![IMG_0040](https://github.com/user-attachments/assets/bd0ec6ea-bafc-4d4b-96dc-78d4d41b6df3)
![Screenshot_20250201-132553](https://github.com/user-attachments/assets/9793f8b2-75d3-49c2-a9a0-ea4ad57e1b83)

**Reply by Manos1966** [reply]
erkenne ich einen Cerbo angeschlossen? ueber VE.Direkt? 
hmmm ich habe ihn irgendwie "verpasst" auf deiner Beschreibung https://github.com/hoylabs/OpenDTU-OnBattery/discussions/1416#discussioncomment-11992732 🤦‍♂️ 

Deine sechs Paneele haben eindeutig nicht mehr als 796W geliefert.
![image](https://github.com/user-attachments/assets/41057ccf-01e0-44c7-aba2-a3cdbffb4227)

**Reply by gruenteefan** [reply]
> erkenne ich einen Cerbo angeschlossen? ueber VE.Direkt? hmmm ich habe ihn irgendwie "verpasst" auf deiner Beschreibung [#1416 (comment)](https://github.com/hoylabs/OpenDTU-OnBattery/discussions/1416#discussioncomment-11992732) 🤦‍♂️
> 
> Deine sechs Paneele haben eindeutig nicht mehr als 796W geliefert. ![image](https://private-user-images.githubusercontent.com/129660360/409311983-41057ccf-01e0-44c7-aba2-a3cdbffb4227.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzg2OTA4MjMsIm5iZiI6MTczODY5MDUyMywicGF0aCI6Ii8xMjk2NjAzNjAvNDA5MzExOTgzLTQxMDU3Y2NmLTAxZTAtNDRjNy1hYmEyLWEzY2RiZmZiNDIyNy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMjA0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDIwNFQxNzM1MjNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wZDEzM2UyNzUwMWFlNGI5YjhhZjc1NWRkNWZlMjk2MzU3N2FlYzc1ZjQxODI1NDFjOGUwZjFjMDMwNDk0NWY0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.12H65ksn4a5_BtpPTgDfkYIhi8vAePO43z4edgsko5w)



Ja, das ist ein Cerbo GX. Angeschlossen über BMS-CAN am Pylontech und (da der VE.direct-Anschluss an die DTU nicht funktioniert) am Victron MPPT.

Heute habe ich von Lukas den unteren Teil der Platine erhalten, um auszuschließen, dass auf meiner selbigen ein Fehler ist. Ich habe den ESP umgesteckt. Leider Ernüchterung. Sobald ich den ve.Direct-Stecker einstecke, geht die Akkuladung aus, nach ein paar Sekunden erlischt die Leuchte am MPPT und der Hoymiles geht aus bzw. auf Störung. Kann das an einer fehlerhaften Einstellung am Victron Laderegler liegen? Das ist schön langsam echt zum Verzweifeln. 

 Eine Alternative zur VE-Direct-Schnittstelle wäre ja auch noch der Abgriff der MPPT-Daten über MQTT. Leider habe ich im MQTT-Explorer hier nicht die erforderlichen Datenpunkte (Ausgangsleistung Solarladeregler V und A). 

Habt ihr hier noch einen Tipp für mich?

**Reply by knopers1** [reply]
ja, habe ich....
die Display Schnittstelle kannst Du für den Victron nutzen. Damit hast du keinen Levelshifter dazwischen!
folgendes: Du nimmst die Schnittstelle für das Display zum Anschluss. Du änders die pin_mapping um. Victron kommt am Anschlusspunkt für das Display

Type-->gpio9-->RX
DATA-->gpio11-->TX

und GND untereinander zwingend verbinden! Damit bekommst du den Victron sicherlich zum laufen!




"display": {
            "type": 9, <-- da auf -1 ändern
            "data": 11,<-- da auf -1 ändern
            "clk": 10

"victron": {
            "rx": 9,
            "tx": 11,





![408211216-c841b70d-f283-49eb-8c02-2bc4c120aba9](https://github.com/user-attachments/assets/c4430a7a-ad02-4abb-8497-b59f5f610af1)

**Reply by gruenteefan** [reply]
@knopers1 
Super, das funktioniert jetzt! Kannst Du mir bitte mal ausdeutschen, warum das mit meinem 150/35 am Display-Anschluss funktioniert und an dem separaten VE.direct-Anschluss nicht?

Ich danke euch allen sehr!

**Reply by knopers1** [reply]
Tja, das ist eine gute Frage..... Die Schaltung macht für mich keinen Sinn. Ich habe das PCB vom Lukas ebenfalls nicht ans laufen bekommen! Exakt brach die DTU zusammen, nachdem der Victron zugesteck worden ist. Über die Display Schnittstelle lief es auf anhieb. 
Die Dokumentaion macht mich irre... 
Da die 5V des Victron auf den VDD2 geschaltet sind, muss die VE-Direct des Victron an VIA (TX) und VOB (RX) geschaltet sein.
Die ESP32 Seite ---> GPIO46 (RX) --> VOA Pin2
                                 GPIO45 (TX) --> VIB Pin3
VDD1 ---> 3,3V des ESP´s...


![1](https://github.com/user-attachments/assets/f1ce43d2-8549-45ea-853c-9df283621859)

Dazu gibt es diesen Stromlaufplan!  VOA und VIB für den GPIO 45 ung GPIO46 passen noch, Rest der Zeichnung ist für den Popo...

![2](https://github.com/user-attachments/assets/fb8be06e-9d11-470f-8b1c-647a5d97762d)

**So müsste es aussehen!**

![2](https://github.com/user-attachments/assets/4a07d5f6-1d47-449e-9cb4-3cc6427cc3f8)

Die Frage jetzt, wie das PCB vom Lukas wirklich verdrahtet ist. Bei mir brachte die DTU zusammen, nachdem der Victron über die beiden VE-Direct Schnittstelle verbunden wurde. Das kann doch kein Zufall sein!

Des Weiteren aus meiner Erfahrung heraus, lief ein Victron 100/20 nur über die TX Schnittstelle ohne Probleme, während ein Victron 150/35 beide Signale (RX und TX) nötig hat. 

Victron100/20 an ADUM1201
![2](https://github.com/user-attachments/assets/289d04d0-48f9-4991-ab3e-15109794f1a6)

Mal angenommen, dass das PCB exakt so aufgebaut ist wie der Stromlaufplan gezeichnet ist, so könnte man damit einen Victron 100/20 betreiben. Ein Victron 150/35 oder ein Victron 250/60 dürfte aber nicht funktionieren!
**Damit ginge nur die TX des >Victrons über den Adum1201, und die RX direckt auf den ESP32,....** Was ein Unsinn an der Stelle ...

EDIT: Jetzt ist alles klar! Der Stromlaufplan ist richtig, und das PCB exakt danach verdrahtet!
dort folgendes gefunden!
https://github.com/hoylabs/OpenDTU-OnBattery/discussions/360


![2](https://github.com/user-attachments/assets/98e1ee17-8869-44f8-ae74-e06ac2fea535)


Da haben wir die Info, weshalb das PCB nur an einem Victron 100/20 funktionieren kann, eventuell noch Victron 75/10 oder 75/15! **Damit kann aber KEIN Victron 150/35 oder ein Victron 250/60 laufen !.** 

@LukasVFL99 
Kannst Du da etwas Licht bringen? Ich bahupte Mal, dass dein PCB wirklich nur mit den "kleinen" Victrons laufen kann, diese Info hätte ich bei der Bestellung gerne von Dir gehört!
